examples/probability.dx

```diff
@@ -0,0 +1,297 @@
+
+' # Mini-PPL for Discrete variables in Dex
+
+' This document is an example for creating a small PPL in Dex. This is
+inspired by PPL's written for Haskell that utilize monadic
+represtations of the chain rule for representing discrete
+variables. Notably though in Dex we aim for a non-monadic style that
+primarily uses tables and autodifferentiation. 
```

Comment:
nit: autodifferentiation -> "automatic differentiation" or "autodiff". The current form reads a bit weird to me?